File: guests/ocaml/format.ml
(* The formatter door and the catalog, OCaml port — guests/rust/format.rs,
   tools/scenes/format.steps. *)

open Kaya_app

let () =
  let app = Kaya_app.create () in
  catalog "format";
  let d = { year = 2026; month = 9; day = 7 } in
  let t = { hour = 8; minute = 30 } in

  build app (fun () ->
     (* Fourteen labels and a row: taller than the default window. *)
     window ~title:"format" ~width:540.0 ~height:560.0 ();
     let root =
       column
         [
           label ~text:(Fmt.date ~length:`Short d) (* label#0 *);
           label ~text:(Fmt.date ~length:`Medium d) (* label#1 *);
           label ~text:(Fmt.date ~length:`Long d) (* label#2 *);
           label ~text:(Fmt.time ~length:`Short t) (* label#3 *);
           label ~text:(Fmt.date_time ~length:`Medium d t) (* label#4 *);
           label ~text:(Fmt.number 1234567.891) (* label#5 *);
           label ~text:(Fmt.percent 0.256) (* label#6 *);
           label ~text:(Fmt.currency 1234567.89 "USD") (* label#7 *);
           label ~text:(tr "items" [ "count", `Int 1 ]) (* label#8 *);
           label ~text:(tr "items" [ "count", `Int 3 ]) (* label#9 *);
           label ~text:(tr "greeting" [ "name", `Text "Ada" ]) (* label#10 *);
           row
             [
               label ~text:"first" (* label#11 *);
               spacer ~grow:1.0;
               label ~text:"last" (* label#12 *);
             ]
           (* row#0 *);
           label ~text:(Fmt.locale ()).tag (* label#13 *);
         ]
         ()
     in
     mount root);

  exit (run app)

File: guests/swift/format.swift
// The format scene, Swift port — guests/rust/format.rs, tools/scenes/format.steps
// (and formatde, formatar through KAYA_LOCALE).

import Foundation
import Kaya

KayaApp.run { app in
    KayaApp.catalog("format")
    let d = KayaDate(year: 2026, month: 9, day: 7)
    let t = KayaTime(hour: 8, minute: 30)
    app.build { tx in
        tx.window(title: "format", width: 540, height: 560)
        let root = tx.column { root in
            tx.label(bind: tx.signal(.str(KayaApp.fmt.date(d, .short))))  // label#0
            tx.label(bind: tx.signal(.str(KayaApp.fmt.date(d, .medium))))  // label#1
            tx.label(bind: tx.signal(.str(KayaApp.fmt.date(d, .long))))  // label#2
            tx.label(bind: tx.signal(.str(KayaApp.fmt.time(t, .short))))  // label#3
            tx.label(bind: tx.signal(.str(KayaApp.fmt.dateTime(d, t, .medium))))  // label#4
            tx.label(bind: tx.signal(.str(KayaApp.fmt.number(1234567.891))))  // label#5
            tx.label(bind: tx.signal(.str(KayaApp.fmt.percent(0.256))))  // label#6
            tx.label(bind: tx.signal(.str(KayaApp.fmt.currency(1234567.89, "USD"))))  // label#7
            tx.label(bind: tx.signal(.str(KayaApp.tr("items", ["count": 1]))))  // label#8
            tx.label(bind: tx.signal(.str(KayaApp.tr("items", ["count": 3]))))  // label#9
            tx.label(bind: tx.signal(.str(KayaApp.tr("greeting", ["name": "Ada"]))))  // label#10
            tx.row { _ in
                tx.label(bind: tx.signal(.str("first")))  // label#11
                tx.spacer()
                tx.label(bind: tx.signal(.str("last")))  // label#12
            }  // row#0
            tx.label(bind: tx.signal(.str(KayaApp.fmt.locale().tag)))  // label#13
            return root
        }
        tx.mount(root)
    }
}
